### DISCUSSION TITLE
wrapper for vectorized functions?

### DISCUSSION THREAD

**Discussion by adriesse** [lmfit/uncertainties#144] [Ideas]
The `wrap` function currently only handles scalar arguments.  Would it be hard to make it work on arrays?

**Comment by lebigot** [comment]
Not too much: there is a simple (but slower) method: the first step in the wrap() function would be to convert the numbers from arrays into individual arguments of a function, which would be wrapped by the current wrap() function.

A faster approach would adapt wrap() and mimic the current approach but apply it to each array element individually.

**Comment by adriesse** [comment]
I'm running 100K+ values through so I think making individual arguments would not be an option.  I do have a function and derivative function that are vectorized, so is there potential with an adapted wrap() to call those once with vectors rather than element by element?  I'm afraid I don't understand the code well enough at this point to judge.

Using vectorize() on the result from wrap() at least keeps me going in the meantime.
